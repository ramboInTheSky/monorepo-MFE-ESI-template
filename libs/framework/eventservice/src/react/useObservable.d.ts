import { ESB } from "../esb";
export declare const useCommonObservable: (esb: ESB, callback: (data: any) => void) => void;
