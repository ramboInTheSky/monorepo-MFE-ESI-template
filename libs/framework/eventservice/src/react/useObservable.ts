import {useEffect} from "react"
import {ESB} from "../esb"

export const useCommonObservable = (esb: ESB, callback: (data: any) => void) => {
    return useEffect(() => {
        const {subscription} = esb.subscribe(callback)
        return () => {
            subscription.unsubscribe()
        }
    // eslint-disable-next-line react-hooks/exhaustive-deps
    }, [])
}
