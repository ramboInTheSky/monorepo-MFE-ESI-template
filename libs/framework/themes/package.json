{
  "name": "@monorepo/themes",
  "version": "17.0.0",
  "author": "",
  "description": "UI theming",
  "main": "./lib/index.js",
  "module": "./lib/index.es.js",
  "types": "./lib/index.d.ts",
  "files": [
    "lib/**/*"
  ],
  "scripts": {
    "build-lib": "rollup -c && yalc push --safe --replace",
    "build-ci": "rollup -c rollup.config.prod.js",
    "lint": "eslint --ext .tsx,.ts src/",
    "tsc": "tsc",
    "test": "jest src/"
  },
  "devDependencies": {
    "@monorepo/eslint": "^3.3.0",
    "@types/jest": "24.9.1",
    "@types/node": "12.12.28",
    "@typescript-eslint/parser": "2.19.2",
    "jest": "24.9.0",
    "jest-coverage-ratchet": "0.2.3",
    "jest-junit": "10.0.0",
    "ts-jest": "24.3.0",
    "typescript": "3.8.2",
    "yalc": "^1.0.0-pre.53"
  },
  "dependencies": {
    "@mui/material": "^5.2.8",
    "@mui/styles": "^5.2.3",
    "@monorepo/core-logger": "^7.3.0"
  },
  "peerDependencies": {
    "@mui/material": "^5.2.8",
    "@mui/styles": "^5.2.3",
    "@monorepo/core-logger": "^7.3.0"
  },
  "gitHead": "",
  "jest": {
    "preset": "ts-jest"
  }
}
