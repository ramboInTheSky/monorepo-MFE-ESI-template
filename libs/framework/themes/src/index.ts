export * from "./themecolors"
export * from "./fonts"
export * from "./breakpoints"
export * from "./themeprovider"
export * from "./isThemeValid"
export * from "./theme"
export * from "../__mocks__/mockTheme"
