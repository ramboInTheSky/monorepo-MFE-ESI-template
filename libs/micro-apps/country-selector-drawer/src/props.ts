import {Country} from "./models/countryselector"

export type CountrySelectorProps = {
    realm: string
    territory: string
    language: string
    textAlignment: string
    cdnBaseUrl: string
    siteUrl: string
    isInternationalCountry: boolean
}

export interface CountrySelectorInternalProps extends CountrySelectorProps {
    flagIconUrl: string
    showOverlay: boolean
    selectedLanguage: string
    selectedCountry: Country
    loaded: boolean
    ROWLinkUrl: string
    shopNowOnClick: () => void
    openDrawer: (showOverlay?: boolean) => void
    closePanel: () => void
    selectLanguage: (language: string) => void
    selectDefaultCountry: (countryCode: string) => void
    getCountriesListThunk: () => void
    redirectPageToAlternativeLanguage: () => void
    showCountrySelector: boolean
}
