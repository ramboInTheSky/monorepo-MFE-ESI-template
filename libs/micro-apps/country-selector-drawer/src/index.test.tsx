import React from "react"
import {ThemeProvider} from "@mui/material"
import {render} from "@testing-library/react"
import {ThemeProvider as SCThemeProvider} from "styled-components"
import {mockTheme as theme} from "@monorepo/themes"
import {CountrySelectorDrawer} from "."
import {CountrySelectorProps} from "./props"
import muiTheme from "./__mocks__/mockMUITheme"

const props: CountrySelectorProps = {
    realm: "amido",
    territory: "gb",
    language: "en",
    textAlignment: "",
    cdnBaseUrl: "",
    siteUrl: "",
    isInternationalCountry: false,
}

describe("Country Selector Drawer Package - ", () => {
    it("should match the snapshot template", () => {
        const {asFragment} = render(
            <ThemeProvider theme={muiTheme}>
                <SCThemeProvider theme={theme}>
                    <CountrySelectorDrawer {...props} />
                </SCThemeProvider>
            </ThemeProvider>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
