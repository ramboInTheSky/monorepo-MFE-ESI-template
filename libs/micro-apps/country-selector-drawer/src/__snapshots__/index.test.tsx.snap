// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Country Selector Drawer Package -  should match the snapshot template 1`] = `
<DocumentFragment>
  <div
    class="components__Wrapper-sc-1piawcx-1 iKCyuV countrylangselector"
    data-testid="header-country-lang-selector"
  >
    <button
      class="components__Container-sc-1piawcx-5 eGkZmG"
    >
      <img
        alt="gb"
        class="components__CountryFlagImg-sc-1piawcx-7 kAbTiM"
        data-testid="header-country-lang-flag"
        src="/icons/shared/countryflags/gb.png"
      />
    </button>
  </div>
</DocumentFragment>
`;
